src/Type/UnionType.php

```diff
@@ -214,24 +219,29 @@ public function isSuperTypeOf(Type $otherType): TrinaryLogic
 			|| $otherType instanceof ConditionalTypeForParameter
 			|| $otherType instanceof IntegerRangeType
 		) {
-			return $otherType->isSubTypeOf($this);
+			return $otherType->isSubTypeOfWithReason($this);
 		}
 
-		$result = TrinaryLogic::createNo()->lazyOr($this->getTypes(), static fn (Type $innerType) => $innerType->isSuperTypeOf($otherType));
+		$result = AcceptsResult::createNo()->or(...array_map(static fn (Type $innerType) => $innerType->isSuperTypeOfWithReason($otherType), $this->types));
 		if ($result->yes()) {
 			return $result;
 		}
 
 		if ($otherType instanceof TemplateUnionType) {
-			return $result->or($otherType->isSubTypeOf($this));
+			return $result->or($otherType->isSubTypeOfWithReason($this));
 		}
 
 		return $result;
 	}
 
 	public function isSubTypeOf(Type $otherType): TrinaryLogic
 	{
-		return TrinaryLogic::lazyExtremeIdentity($this->getTypes(), static fn (Type $innerType) => $otherType->isSuperTypeOf($innerType));
```

Comment:
Same `lazyExtremeIdentity` into `extremeIdentity`, like it's done for isAcceptedWithReasonBy